core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -434,7 +433,7 @@ public Integer sortOrderId() {
     if (map != null) {
       Map<K, V> copy = Maps.newHashMapWithExpectedSize(map.size());
       copy.putAll(map);
-      return Collections.unmodifiableMap(copy);
+      return copy;
```

Comment:
There is a patch to address the kryo serialization issue: https://github.com/apache/iceberg/pull/2343